llvm/test/Feature/Repo/repo_discardable_if_unused_test.ll

```diff
@@ -0,0 +1,22 @@
+; Check the caller's pruning  if its callee has noinline attribute and may be discarded if the callee is not used.
+;
+; RUN: rm -f %t.db
+; RUN: env REPOFILE=%t.db opt -S -mtriple x86_64-pc-linux-gnu-repo %S/Inputs/repo_discardable_GOs.ll -o %t 2>&1
+; RUN: env REPOFILE=%t.db llc -filetype=obj %t -o /dev/null 2>&1
+; RUN: env REPOFILE=%t.db opt -S -mtriple x86_64-pc-linux-gnu-repo %s | FileCheck %s
```

Comment:
I think that this is a test for **the fix** rather than for the reported issue (callee function being eliminated). Could you extend this to make sure that callee not removed by DCE?

Reply:
Good point, the test is extended.